src/overlays/actors/ovl_En_Insect/z_en_insect.c

```diff
@@ -21,55 +41,751 @@ const ActorInit En_Insect_InitVars = {
+        this->actor.initPosRot.pos.y = thisTemp->soilActor->actor.posRot.pos.y;
+        this->actor.initPosRot.pos.z = (Math_Rand_ZeroOne() - 0.5f) * phi_f2 + thisTemp->soilActor->actor.posRot.pos.z;
+    }
+
+    if (D_80A7DEB0 > 0.999f) {
+        this->unk_328 = Math_Vec3f_Yaw(&this->actor.posRot.pos, &this->actor.initPosRot.pos);
+        this->unk_324 = Math_Rand_ZeroOne() * 0.6f + 0.6f;
+    } else if (Math_Rand_ZeroOne() < 0.07f) {
+        if (this->unk_324 > 1.0f) {
+            this->unk_324 = 0.1f;
+        } else {
+            this->unk_324 = Math_Rand_ZeroOne() * 0.8f + 1.0f;
+        }
+
+        sp34 = 1.3f - D_80A7DEB0;
+        if (sp34 < 0.0f) {
+            sp34 = 0.0f;
+        } else {
+            if (sp34 > 1.0f) {
+                phi_f0 = 1.0f;
+            } else {
+                phi_f0 = sp34;
+            }
+            sp34 = phi_f0;
+        }
+
+        sp38 = (Math_Rand_ZeroOne() - 0.5f) * 65535.0f * sp34;
+        this->unk_328 = Math_Vec3f_Yaw(&this->actor.posRot.pos, &this->actor.initPosRot.pos) + sp38;
+    }
+
+    Actor_SetScale(&this->actor, CLAMP_MAX(thisTemp->actor.scale.x + 0.0008f, 0.01f));
+
+    if (this->actor.bgCheckFlags & 1) {
+        Math_SmoothScaleMaxMinF(&this->actor.speedXZ, this->unk_324, 0.1f, 0.5f, 0.0f);
+        Math_ApproxUpdateScaledS(&this->actor.posRot.rot.y, this->unk_328, 2000);
+        sp50 = Math_ApproxUpdateScaledS(&this->actor.posRot.rot.x, 0, 2000);
+        this->actor.shape.rot.y = this->actor.posRot.rot.y;
+        this->actor.shape.rot.x = this->actor.posRot.rot.x;
+    } else {
+        Math_SmoothScaleMaxMinF(&this->actor.speedXZ, 0.0f, 0.1f, 0.5f, 0.0f);
+        this->actor.speedXZ += (Math_Rand_ZeroOne() - 0.5f) * 0.14f;
+        this->actor.velocity.y += Math_Rand_ZeroOne() * 0.12f;
+        this->actor.posRot.rot.y += this->unk_316;
+        this->actor.shape.rot.y = this->actor.posRot.rot.y;
+        this->actor.shape.rot.x -= 2000;
+    }
+
+    phi_f2 = Math_Rand_ZeroOne() * 0.5f + this->actor.speedXZ * 1.3f;
+    if (phi_f2 < 0.0f) {
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+    } else {
+        if (phi_f2 > 1.9f) {
+            phi_f0 = 1.9f;
+        } else {
+            phi_f0 = phi_f2;
+        }
+        this->skelAnime.animPlaybackSpeed = phi_f0;
+    }
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (!(this->unk_314 & 0x40) && this->unk_314 & 1 && this->actor.bgCheckFlags & 1) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_MUSI_LAND);
+        this->unk_314 |= 0x40;
+    }
+
+    if (sp3A == 2 && this->unk_314 & 0x10 && !(this->unk_314 & 0x80)) {
+        if (this->unk_32A >= 15) {
+            if (this->soilActor != NULL) {
+                temp_a0 = ((this->soilActor->actor.params >> 8) & 0x1F) - 1;
+                temp_a1 = temp_a0 & 3;
+
+                if ((((u32)(((s32*)gSaveContext.gsFlags)[temp_a0 >> 2] & D_8012723C[temp_a1]) >> D_8012724C[temp_a1]) &
```

Comment:
This needs to be cleaned up. The save context struct should be updated to avoid casting to a different type.